### PR TITLE
Added cascaded shadow maps documentation

### PR DESCRIPTION
**Pull request by mvaligursky**
Fixes https://github.com/playcanvas/developer.playcanvas.com/issues/307
Screenshot 2022-04-19 at 15 04 22
Screenshot 2022-04-19 at 15 04 31

### CODE REVIEW

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 1]
```diff
@@ -18,9 +18,23 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
 
 ![Model Component][6]
 
-Now you need to specify which graphical objects in your scene cast and receive shadows. By default, all model components receive shadows but do not cast them. Therefore, in order to see an entity cast shadows, select it in the Hierarchy, locate the model component in the Inspector and check the 'Cast Shadows' option.
+Now you need to specify which graphical objects in your scene cast and receive shadows. By default, all render and model components cast and receive shadows. To modify these properties, select the entity in the Hierarchy, locate the render or model component in the Inspector and uncheck the 'Cast Shadows' or 'Receive Shadows' option as required.
```
Comment (yaustar) [inline content/en/user-manual/graphics/lighting/shadows.md:21]: Optional: Would be great to have a screenshot of these options

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 2]
```diff
@@ -18,9 +18,23 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
...
 
-Shadows should now be visible in the Editor's Viewport.
+## Shadow Cascades
+
+When a directional shadow is used over a large area, it often exhibits aliasing, where a shadow near the camera has a low resolution. Capturing the shadow in a single shadow map requires very high and impractical resolution to improve this.
+
+Shadow cascades help to fix this problem by splitting the camera view frustum along the viewing direction, and a separate shadow map is used for each split. This gives nearby objects one shadow map, and another shadow map captures everything in the distance, and optionally additional shadow maps in between.
+
+The following properties can be used to set up shadow cascades.
+
```
Comment (yaustar) [inline content/en/user-manual/graphics/lighting/shadows.md:30]: Screenshots to compare with would be useful for users to tell the difference and know whether they should use it or not

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 3]
```diff
@@ -18,9 +18,23 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
...
+### Number of cascades
+
+Number of cascades represents the number of view frustum subdivisions, and can be 1, 2, 3 or 4. The default value of 1 represents a single shadow map.
+
+### Distribution of cascades
+
+The distribution of subdivision of the camera frustum for individual shadow cascades. A value in the range of 0 to 1 can be specified. Value of 0 represents a linear distribution, value of 1 represents a logarithmic distribution. Larger value increases the resolution of the shadows in the near distance.
```
Comment (yaustar) [inline content/en/user-manual/graphics/lighting/shadows.md:37]: Why should a user use a smaller value if a larger value implies better results of increased resolution?

Would screenshots of the different values help here?

File: content/en/user-manual/graphics/lighting/shadows.md (hunk 1 above)
Comment (mvaligursky) [reply content/en/user-manual/graphics/lighting/shadows.md:21]: There is a screenshot for this showing the model component. I just slightly improved the text to have up to date defaults.

File: content/en/user-manual/graphics/lighting/shadows.md (hunk 3 above)
Comment (mvaligursky) [reply content/en/user-manual/graphics/lighting/shadows.md:37]: This is hard to show using a screenshot. I'm updating text to explain better.

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 4]
```diff
@@ -18,9 +18,29 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
...
+
+A screenshot showing a single shadow cascade.
+![One cascade][7]
+
+A screenshot showing four shadow cascades.
+![Four cascades][8]
+
+### Distribution of cascades
+
+The distribution of subdivision of the camera frustum for individual shadow cascades. A value in the range of 0 to 1 can be specified. Value of 0 represents a linear distribution, value of 1 represents a logarithmic distribution. Visually, a higher value distributes more shadow map resolution to foreground objects, while lower value distributes it to more distant objects.
```
Comment (willeastcott) [inline content/en/user-manual/graphics/lighting/shadows.md:43]:
```suggestion
The distribution of subdivision of the camera frustum for individual shadow cascades. A value in the range of 0 to 1 can be specified. A value of 0 represents a linear distribution and a value of 1 represents a logarithmic distribution. Visually, a higher value distributes more shadow map resolution to foreground objects, while a lower value distributes it to more distant objects.
```

**Review by willeastcott — APPROVED** [review]

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 5]
```diff
@@ -18,9 +18,29 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
...
+Shadow cascades help to fix this problem by splitting the camera view frustum along the viewing direction, and a separate shadow map is used for each split. This gives nearby objects one shadow map, and another shadow map captures everything in the distance, and optionally additional shadow maps in between.
+
+The following properties can be used to set up shadow cascades.
+
+### Number of cascades
+
+Number of cascades represents the number of view frustum subdivisions, and can be 1, 2, 3 or 4. The default value of 1 represents a single shadow map.
+
+A screenshot showing a single shadow cascade.
+![One cascade][7]
```
Comment (willeastcott) [inline content/en/user-manual/graphics/lighting/shadows.md:36]:
```suggestion
A screenshot showing a single shadow cascade.

![One cascade][7]
```

File: content/en/user-manual/graphics/lighting/shadows.md [hunk 6]
```diff
@@ -18,9 +18,29 @@ By default, shadow casting is disabled in PlayCanvas. You have to explicitly ena
...
+
+A screenshot showing four shadow cascades.
+![Four cascades][8]
```
Comment (willeastcott) [inline content/en/user-manual/graphics/lighting/shadows.md:39]:
```suggestion
A screenshot showing four shadow cascades.

![Four cascades][8]
```

**Comment by willeastcott** [comment]
In case you're wondering why I put a new line between the text and the image, it's kind of an unwritten coding standard for our Markdown. The motive is that, when uploaded to Transifex, the text line and image line will be treated as two separate paragraphs to translate. And obviously, the image line doesn't need translation.

**Comment by mvaligursky** [comment]
Nice, thanks @willeastcott - I didn't know that.

**Comment by yaustar** [comment]
If 4 subdivisions looks good, why use 2 or 3?

**Comment by mvaligursky** [comment]
> If 4 subdivisions looks good, why use 2 or 3?

added `Note that the number of shadow cascades has an effect on performance, as each shadow casting mesh might need to be rendered into more than a single shadow map.`, thanks!

**Pull request merged**